### PR TITLE
R doc 1641 cluster certificate exec

### PR DESCRIPTION
**Pull request by morhilai**

### CODE REVIEW

File: Documentation/4.2/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown [hunk 1]
```diff
@@ -4,7 +4,7 @@ In RavenDB, configuration values can be set using environment variables, command
 
 {NOTE This section explains how to setup authentication **manually**. Please also take a look at the automated [Setup Wizard](../../../start/installation/setup-wizard) which lets you setup authentication in a much easier and faster way. /}
 
-To enable authentication, either `Security.Certificate.Path` or `Security.Certificate.Exec` must be set in [settings.json](../../configuration/configuration-options#json).
+To enable authentication, either `Security.Certificate.Path` or `Security.Certificate.Load.Exec` must be set in [settings.json](../../configuration/configuration-options#json).
```
Comment (iftahbe) [inline Documentation/4.2/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown]: Let's add a comment either here or in the FAQ section with this answer: https://groups.google.com/forum/#!searchin/ravendb/exec%7Csort:date/ravendb/9KcfDtOF43U/VC7CTCu8EQAJ
What to do ff the user doesn't want to use the new scripts.

File: Documentation/4.2/Raven.Documentation.Pages/server/configuration/security-configuration.markdown [hunk 2]
```diff
@@ -27,31 +27,88 @@ The (optional) password of the .pfx certificate file.
...
+Deprecated in 4.2 and replaced with [Security.Certificate.Load.Exec.Arguments](../../server/configuration/security-configuration#security.certificate.load.exec.arguments), see below.
 
-- **Type**: `string`
-- **Default**: `null`
-- **Scope**: Server-wide only
+{PANEL/}
+
+{PANEL:Security.Certificate.Load.Exec}
+
+A command or executable that returns a .pfx cluster certificate when invoked by RavenDB. If specified, RavenDB will use HTTPS/SSL for all network activities.
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/configuration/security-configuration.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/configuration/security-configuration.markdown [hunk 3]
```diff
@@ -27,31 +27,88 @@ The (optional) password of the .pfx certificate file.
...
+
+- **Type**: `string`  
+- **Default**: `null`  
+- **Scope**: Server-wide only  
+
+{PANEL/}
+
+{PANEL:Security.Certificate.Change.Exec}
+
+A command or executable that replaces the old cluster certificate with the renewed cluster certificate. When invoked, the cluster leader node will send the
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/configuration/security-configuration.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    "Security.Certificate": {
+        "Path": "/home/secrets/server.pfx",
+        "Password": "s3cr7t p@$$w0rd"
+    }
+} 
+{CODE-BLOCK/}
+
+The second way to enable authentication is to set `Security.Certificate.Load.Exec`. 
+
+This option is useful when you want to protect your certificate (private key) with other solutions such as "Azure Key Vault", "HashiCorp Vault" or even Hardware-Based Protection. RavenDB will invoke a process you specify, so you can write your own scripts / mini programs and apply whatever logic you need. It creates a clean separation between RavenDB and the secret store in use.
```
Comment (ayende) [inline Documentation/4.2/Raven.Documentation.Pages/server/security/authentication/certificate-configuration.markdown]:
```suggestion
This option is useful when you want to protect your certificate (private key) with other solutions such as "Azure Key Vault", "HashiCorp Vault" or even Hardware Security Module (HSM). RavenDB will invoke a process you specify, so you can write your own scripts / mini programs and apply whatever logic you need. It creates a clean separation between RavenDB and the secret store in use.
```

**Comment by arekpalinski** [comment]
@morhilai I'm merging this. Please check Oren's comment https://github.com/ravendb/docs/pull/1083#pullrequestreview-297937714

If needed create a dedicated issue or reopen this one

**Pull request merged**